tsinfer/inference.py

```diff
@@ -230,7 +244,7 @@ def infer(
         engine=engine,
         num_threads=num_threads,
         recombination_rate=recombination_rate,
-        mismatch_rate=mismatch_rate,
+        # mismatch_ratio=mismatch_ratio,
```

Comment:
Looks like ``mismatch_ratio`` is commented out here?

Reply:
That's because I don't currently allow it to be specified in the call to `infer()`; it's restricted to `.match_samples()` and `match_ancestors()` - see my point 2 in the intro. Do you think we should allow it in `infer()`, and then set the same mismatch_ratio in both phases of the algorithm?

Reply:
I have flip-flopped on this and now put mismatch_ratio back into `infer()`. But if you want to specify the full probability arrays you can't use `infer()` - you have to do the 2 matching stages separately.